test/functional/test_framework/test_node.py

```diff
@@ -140,6 +142,29 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             AddressKeyPair('mzRe8QZMfGi58KyWCse2exxEFry2sfF2Y7', 'cPiRWE8KMjTRxH1MWkPerhfoHFn5iHPWVK5aPqjW8NxmdwenFinJ'),
     ]
 
+    def is_valgrind_present(self):
+        """Check for valgrind presence"""
+        try:
+          version = subprocess.run(["valgrind", "--version"], stdout=subprocess.PIPE).stdout.decode("utf-8")
+          if version.find("valgrind", 0, 10) > -1:
+              return True
+          else:
+              return False
+        except OSError:
+          return False
+
+    def should_exit_valgrind_on_first_error(self):
+        """Determine if valgrind should run with --exit-on-first-error flag (>=3.15 runs with it while older versions don't)"""
```

Comment:
I believe this was added in 3.14: https://stackoverflow.com/a/50289059

Reply:
Will adapt!

Reply:
Will then simply produce an empty string if the flag should not be active, because it's "no" anyway for versions that support it.

Like: **"--exit-on-first-error=yes" if exit_on_first_error else ""**

Reply:
Yes, I have valgrind-3.14.0 installed on debian, and valgrind --help does have the option for `--exit-on-first-error=no|yes`.

Reply:
Yes, I have valgrind-3.14.0 installed on debian, and valgrind --help does have the option for `--exit-on-first-error=no|yes`. Will test the updated version of this PR!

Reply:
Just force pushed changes. :)